sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -357,70 +387,179 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
+    }
+    const offeringValidationModel = new OfferingValidationModel();
+    offeringValidationModel.offeringName = offering.name;
+    offeringValidationModel.studyStartDate = offering.studyStartDate;
+    offeringValidationModel.studyEndDate = offering.studyEndDate;
+    offeringValidationModel.actualTuitionCosts = offering.actualTuitionCosts;
+    offeringValidationModel.programRelatedCosts = offering.programRelatedCosts;
+    offeringValidationModel.mandatoryFees = offering.mandatoryFees;
+    offeringValidationModel.exceptionalExpenses = offering.exceptionalExpenses;
+    offeringValidationModel.offeringDelivered =
+      offering.offeringDelivered as OfferingDeliveryOptions;
+    offeringValidationModel.lacksStudyBreaks = offering.lacksStudyBreaks;
+    offeringValidationModel.offeringType = offering.offeringType;
+    offeringValidationModel.programContext = offering.educationProgram;
+    offeringValidationModel.locationId = offering.institutionLocation.id;
+    offeringValidationModel.offeringIntensity = offering.offeringIntensity;
+    offeringValidationModel.yearOfStudy = offering.yearOfStudy;
+    offeringValidationModel.showYearOfStudy = offering.showYearOfStudy;
+    offeringValidationModel.hasOfferingWILComponent =
+      offering.hasOfferingWILComponent as WILComponentOptions;
+    offeringValidationModel.offeringWILComponentType = offering.offeringWILType;
+    offeringValidationModel.offeringDeclaration = offering.offeringDeclaration;
+    offeringValidationModel.offeringType = offering.offeringType;
+    offeringValidationModel.courseLoad = offering.courseLoad;
+    offeringValidationModel.studyBreaks = offering.studyBreaks.studyBreaks;
+    return offeringValidationModel;
+  }
+
+  /**
+   * Gets an offering with all the expected data to a complete
+   * validation be performed.
+   * @param offeringId offering to have the data selected.
+   * @returns offering with data to be validated.
+   */
+  async getOfferingForValidation(
+    offeringId: number,
+  ): Promise<EducationProgramOffering> {
+    return this.repo.findOne({
+      select: {
+        id: true,
+        name: true,
+        studyStartDate: true,
+        studyEndDate: true,
+        actualTuitionCosts: true,
+        programRelatedCosts: true,
+        mandatoryFees: true,
+        exceptionalExpenses: true,
+        offeringDelivered: true,
+        lacksStudyBreaks: true,
+        offeringType: true,
+        educationProgram: {
+          id: true,
+          programIntensity: true,
+          hasWILComponent: true,
+          deliveredOnSite: true,
+          deliveredOnline: true,
+        },
+        institutionLocation: {
+          id: true,
+        },
+        offeringIntensity: true,
+        yearOfStudy: true,
+        showYearOfStudy: true,
+        hasOfferingWILComponent: true,
+        offeringWILType: true,
+        offeringDeclaration: true,
+        courseLoad: true,
+        // TODO: change studyBreaks to the unknown type.
```

Comment:
what is the unknown scenario. for my info

Reply:
For the JSON entity properties, we should use the `unknown` to do the mapping on the entity model otherwise the Typeorm object `select` does not work properly.

Reply:
oh... unknown type. i got it